### DISCUSSION TITLE
Comparing - forever ...?

### DISCUSSION THREAD

**Discussion by BrJohan** [WinMerge/winmerge#573] [General]
Some 30 hours ago I started a comparison between two directories. One contains ~235000 files in ~38000 folders (102 GB). The other contains ~16000 files in ~19000 folders (many empty) (12GB). All compared data reside on two USB3-connexted disks on a Win10 machine (32GB RAM, Core i9)

Rather soon after starting the comparison, the comparison results window tells me:
     Comparing items ...
      Items total  348296
      Items compared  48    <name of a file>

      Buttons "Pause" and "Stop" are clickable..

Now - after all those hours - the comparison results window has not changed. Task Manager shows (for WinMerge 2.16.8)
     CPU: 0 - 0.4 %, Memory ~3.3  MB, Disk: 0%, Network 0%.

Clicking on "Pause", allows me to "Continue". Continue selected.
Clicking on "Stop" gives no response.
Clicking on Close (upper right) allows me to "close the window? - Yes or No". "No" selected.

Is such a long wait for WinMerge to complete an execution like this, reasonable?

**Comment by BrJohan** [comment]
24 hours later: WinMerge still running. Same low CPU usage. Memoryusage  3.0-3.3 MB. No change in GUI..

Is WinMerge behaving well?

**Comment by BrJohan** [comment]
After waiting 105 hours for WinMerge to complete the task I gave it, I give up. Looking for alternative ...

**Reply by reallybeingme** [reply]
I am also having the same problem Winmerge 2.16.9.0 x64. Comparing two directories from two different HDD (179,948 items) and it appears winmerge has stalled out. In task manager it using 0 -  0.1 CPU, 16.7 MB memory,  0 m/s disk.  There is nothing to show that winmerge is still working or is there someway to check it. Or is this a bug?

**Comment by sdottaka** [comment]
If you want to compare a large number of files, I recommend using the binary comparison method.

Could you try that comparison method?

![image](https://user-images.githubusercontent.com/98126/106358166-b4177580-634d-11eb-90b2-d925fc88d577.png)

**Comment by reallybeingme** [comment]
That worked.

Thanks Jack
